### PR TITLE
LRQA 69446

### PR DESCRIPTION
**Pull request by guivilarim**
Related Issue: https://issues.liferay.com/browse/LRQA-69446

- LRQA-69446 Add tests stubs
- LRQA-69446 Create BUTTON_LABEL_INDEXED and SELECTOR_INPUT_INDEXED paths
- LRQA-69446 Create inputDataInUploadField and assertDataInUploadField macros
- LRQA-69446 Add CanInputData test
- LRQA-69446 Fix CanClearImage test
- LRQA-69446 Create assertUploadNotPresent macro
- LRQA-69446 Add CanClearInput test
- LRQA-69446 Add CanNotPublishBlankRequiredField test
- LRQA-69446 Add CanRemoveDuplicatedField test
- LRQA-69446 Add CanInputDataOnDuplicatedField test
- LRQA-69446 Fix CanClearInput test

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by guivilarim** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEImageField.testcase [hunk 1]
```diff
@@ -115,7 +115,7 @@ definition {
 
 		WebContentNavigator.gotoEditCP(webContentTitle = "Web Content Title");
 
-		DERenderer.clearImageField(fieldLabel = "Image");
+		DERenderer.clearField(fieldLabel = "Image");
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEImageField.testcase:118]: @guivilarim, why did you decide to fix a test inside a pull request intended to create new tests only? Was this fix mandatory to the tests being created by this pull succeed? If not, honestly, I believe that we should create a specific pull to tackle the test fix.

File: modules/apps/data-engine/data-engine-test/src/testFunctional/macros/DERenderer.macro [hunk 2]
```diff
@@ -465,15 +465,4 @@ definition {
...
-		if (!(isSet(index))) {
-			var index = "1";
-		}
-
-		AssertTextEquals(
-			key_fieldLabel = "${fieldLabel}",
-			locator1 = "TextInput#SELECTOR_INPUT",
-			value1 = "${webContentUpload}");
-	}
-
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/macros/DERenderer.macro]: @guivilarim, the commit message is inconsistent with the commit content, can you please revisit this commit?

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEUploadField.testcase [hunk 3]
```diff
@@ -26,6 +26,46 @@ definition {
 		}
 	}
 
+	@description = "This is a stubbed description"
+	@priority = "5"
+
+	test {
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEUploadField.testcase]: @guivilarim, where is the test name??

**Review by natocesarrego — CHANGES_REQUESTED** [review]
@guivilarim, I'm able to see an evolution from your previous pull though I also realized some unnecessary content being committed. Please visit my review comments and get back to me if you have ANY question. If you also need some assistance regarding Git, please let me know and I'll be glad to help you. :wink: 

Thanks!

**Comment by natocesarrego** [comment]
@guivilarim, let's talk about this pull today after our daily. :wink: Please remind me about it at the end of our meeting.

File: modules/apps/data-engine/data-engine-test/src/testFunctional/paths/DataEngineRenderer.path [hunk 4]
```diff
@@ -55,6 +55,16 @@
...
 	<td></td>
 </tr>
+<tr>
+	<td>BUTTON_LABEL_INDEXED</td>
+	<td>xpath=(//div[./label[normalize-space(text())='${key_fieldLabel}']]//button[normalize-space()='${text}'])[${index}]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SELECTOR_INPUT_INDEXED</td>
+	<td>xpath=(//label[normalize-space(text())='${key_fieldLabel}']/following-sibling::div//div[contains(@class,'input-group')]//input[not(contains(@type,'hidden'))])[${index}]</td>
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/paths/DataEngineRenderer.path:65]: @guivilarim, the variables `text` and `index` are still missing the prefix `key_`, can you please add it?

Thanks

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEUploadField.testcase (hunk 3 above)
Comment (natocesarrego) [reply modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DEUploadField.testcase]: @guivilarim, have you had the chance to see this comment? Can you please edit this commit to make it clearer to the reviewer?

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**